From 85fd06356510761f652e066e0ff851cd6e8c5f2e Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 18 Apr 2021 17:12:43 +0300
Subject: [PATCH] [SW-28013] pci: adding reset sources condition for iatu init

Software reset and watchdog reset do not affect the iATU
configuration, so there is no need to initialize iATU after those.

Change-Id: I37a6a8e685fccfae17c3b6f68db681d7711b6a50
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_pci.c b/preboot/module_pci.c
index 1bba904..195cf4c 100644
--- a/preboot/module_pci.c
+++ b/preboot/module_pci.c
@@ -6,6 +6,7 @@
 
 #include <pcie_aux_regs.h>
 #include <pcie_core_regs.h>
+#include <psoc_global_conf_regs.h>
 
 #include <bitops.h>
 #include <preboot.h>
@@ -94,6 +95,7 @@
 void module_pci_run(void)
 {
 	bool pci_mem_enabled;
+	unsigned int rst_src;
 
 	if (pci_sts.flr_reg_val == 1) {
 		regmap_writel(1, mmPCIE_AUX_PF_FLR_ACTIVE);
@@ -122,7 +124,8 @@
 	pci_mem_enabled = hl_pci_is_mem_enabled();
 	if (pci_mem_enabled != pci_sts.mem_enabled) {
 		pci_sts.mem_enabled = pci_mem_enabled;
-		if (pci_mem_enabled)
+		rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
+		if (pci_mem_enabled && (rst_src < HL_WD_RST))
 			hl_pci_iatu_init();
 	}
 }
